data/convert.go

```diff
@@ -53,7 +53,10 @@ func NewWith(convert StructOptions, value interface{}) Value {
 	case reflect.String:
 		return String(v.String())
 	case reflect.Slice:
-		var slice []Value
+		if v.IsNil() {
+			return Null{}
```

Comment:
Why does your application want []string => nil?  Conceptually speaking, a nil slice in Go seems much closer to [] in json than null.